controller/src/main/java/org/jboss/as/controller/SimpleResourceDefinition.java

```diff
@@ -64,10 +64,11 @@
      * @param pathElement         the path. Can be {@code null}.
      * @param descriptionProvider the description provider. Cannot be {@code null}
      * @throws IllegalArgumentException if {@code descriptionProvider} is {@code null}.
-     * @deprecated
+     * @deprecated Use {@link #SimpleResourceDefinition(Parameters)}
      */
     @Deprecated
     public SimpleResourceDefinition(final PathElement pathElement, final DescriptionProvider descriptionProvider) {
+        //Can be removed when we get to 3.0.0
```

Comment:
I think we should deprecate it in 1.x first so we can remove it in 2.x already. otherwise we will be stuck with it for 10 years.
we haven't done 1.0.0.Final yet, so it is not too late.